src/System.Linq/src/System/Linq/Where.cs

```diff
@@ -161,24 +194,25 @@ public override Iterator<TSource> Clone()
                 return new WhereArrayIterator<TSource>(_source, _predicate);
             }
 
+            public int GetCount(bool onlyIfCheap) => onlyIfCheap ? -1 : EnumerableHelpers.Count(this);
```

Comment:
Rather than calling EnumerableHelpers.Count, couldn't we just iterate ourselves here, which would avoid the enumerator allocation, the interface calls, etc.?  Same for the WhereListIterator case.